platform/android/export/export.cpp

```diff
@@ -255,134 +256,177 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 		EditorProgress *ep;
 	};
 
+	Vector<String> plugins_filenames;
+	volatile bool plugins_changed;
+	Mutex *plugins_lock;
 	Vector<Device> devices;
 	volatile bool devices_changed;
 	Mutex *device_lock;
-	Thread *device_thread;
+	Thread *check_for_changes_thread;
 	volatile bool quit_request;
 
-	static void _device_poll_thread(void *ud) {
+	static void _check_for_changes_poll_thread(void *ud) {
 
 		EditorExportPlatformAndroid *ea = (EditorExportPlatformAndroid *)ud;
 
 		while (!ea->quit_request) {
 
-			String adb = EditorSettings::get_singleton()->get("export/android/adb");
-			if (FileAccess::exists(adb)) {
+			// Check for plugins updates
+			{
+				// Nothing to do if we already know the plugins have changed.
+				if (!ea->plugins_changed) {
+					String plugins_dir = ProjectSettings::get_singleton()->get_resource_path().plus_file(
+							"android/plugins");
+					if (DirAccess::exists(plugins_dir)) {
 
-				String devices;
-				List<String> args;
-				args.push_back("devices");
-				int ec;
-				OS::get_singleton()->execute(adb, args, true, NULL, &devices, &ec);
-
-				Vector<String> ds = devices.split("\n");
-				Vector<String> ldevices;
-				for (int i = 1; i < ds.size(); i++) {
-
-					String d = ds[i];
-					int dpos = d.find("device");
-					if (dpos == -1)
-						continue;
-					d = d.substr(0, dpos).strip_edges();
-					ldevices.push_back(d);
+						// Check the plugins directory for config (.gdap) files
+						Vector<String> config_files = DirAccess::list_files(plugins_dir, ".gdap");
+
+						ea->plugins_lock->lock();
+
+						if (ea->plugins_filenames.size() != config_files.size()) {
+							ea->plugins_changed = true;
+						} else {
+							for (int i = 0; i < ea->plugins_filenames.size(); i++) {
+								if (ea->plugins_filenames[i] != config_files[i]) {
+									ea->plugins_changed = true;
+									break;
+								}
+							}
+						}
+
+						ea->plugins_lock->unlock();
+					}
 				}
+			}
 
-				ea->device_lock->lock();
+			// Check for devices updates
+			{
+				String adb = EditorSettings::get_singleton()->get("export/android/adb");
+				if (FileAccess::exists(adb)) {
+
```

Comment:
You'll have to run `clang-format` on these changes again as I enabled the rule that disallows such cosmetic empty lines at the start of code blocks.